htdocs/api/class/api_access.class.php

```diff
@@ -194,6 +194,56 @@ public function __isAllowed()
 			// Set also the global variable $user to the $user of API
 			$user = $fuser;
 
+			// set global mysoc after setting conf entity (the entity can be changed with the user logged)
+			// see master.inc.php
+			require_once DOL_DOCUMENT_ROOT.'/societe/class/societe.class.php';
+
+			$fmysoc = new Societe($db);
+			$fmysoc->setMysoc($conf);
```

Comment:
Resetting $mysoc with setMysoc should be done only if we have switched into another entity.
So i think code added should be only just after line 134 and should not be called in any other cases.